lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,49 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  if Puppet::Util::Platform.windows?
+    require 'win32/registry'
+
+    # ensure the symlink folder exists. This also ensures the directory structure
+    # is in place prior to creating the symlink if the folder doesn't exist
+    def set_symlink_folder
+      #TODO:
+      # - Check whether the path contains a symlink #Puppet::FileSystem.symlink?(puppet_modules_vendor_folder)
+      # - To support v3.4.0, not all of these methods existed with Puppet::FileSystem
+      #   Everything was available in 3.5.0 though based
+      #   on https://github.com/puppetlabs/puppet/commit/938b4e7d495f5f15557110040458171e832cecf2
+      puppet_modules_vendor_folder = "#{get_program_files_dir}\\WindowsPowerShell\\Modules\\PuppetVendoredModules"
+      if !Puppet::FileSystem.exist?(puppet_modules_vendor_folder)
+        Puppet::FileSystem.dir_mkpath(puppet_modules_vendor_folder) if !Puppet::FileSystem.dir_exist?(puppet_modules_vendor_folder)
+        Puppet::FileSystem.symlink(vendored_modules_path, puppet_modules_vendor_folder)
```

Comment:
I think you mean `exist?` / `dir_mkpath` /  `dir_exist?` right?

I'm OK with a `3.5` minimum given our feature page says Puppet 3.8 / Puppet 4.0 as the target Puppet versions.

/cc @cowofevil